cont.c

```diff
@@ -3018,6 +3084,41 @@ rb_fiber_scheduler_interface_kernel_sleep(VALUE self)
 {
 }
 
+/*
+ *  Document-method: SchedulerInterface#address_resolve
+ *  call-seq: address_resolve(hostname) -> array_of_stings or nil
+ *
+ *  Invoked by Socket::getaddrinfo and is expected to provide hostname resolution
+ *  in a non-blocking way.
+ *
+ *  The method is expected to return an array of strings corresponding to ip
+ *  addresses the +hostname+ is resolved to, or +nil+ if it can not be resolved.
+ *
+ *  The method support should be considered _experimental_.
+ */
+static VALUE
+rb_fiber_scheduler_interface_address_resolve(VALUE self)
+{
+}
+
+/*
+ *  Document-method: SchedulerInterface#address_resolve
+ *  call-seq: timeout_after(duration, exception_class, *exception_args, &block) -> obj
+ *
+ *  Invoked by Timeout.timeout to perform execution time control in a non-blocking way.
+ *
+ *  The method is expected to execute a +block+, and if its execution takes longer
+ *  than +duration+, to raise +exception_class+ constructed with +exception_args+.
+ *
+ *  If the block is executed successfully, its result should be returned.
+ *
+ *  The method support should be considered _experimental_.
```

Comment:
```suggestion
 *  Document-method: SchedulerInterface#address_resolve
 *  call-seq: timeout_after(duration, exception_class, *exception_arguments, &block) -> result of block
 *
 *  Limit the execution time of a given +block+ to the given +duration+ if
 *  possible. When a non-blocking operation causes the +block+'s execution time
 *  to exceed the specified +duration+, that non-blocking operation should be
 *  interrupted by raising the specified +exception_class+ constructed with the
 *  given +exception_arguments+.
 *
 *  This hook is invoked by Timeout.timeout and can also be invoked directly on
 *  the scheduler.
 *
 *  If the block is executed successfully, its result will be returned.
 *
 *  The exception will typically be raised using Fiber#raise.
```

I don't know if it's valid to reference it using `Fiber#raise`.